### PR TITLE
support for services & methods comments (stubs & runtime inspection)

### PR DESCRIPTION
**Pull request by bjaglin**
Not sure how this can be tested?
* `./make_plugin_proto.sh` shows the impact on enum & enum values
* I verified that the gRPC stubs generated in `e2e` (that do contain comments on RPCs) were coming out as expected:
```scala
  trait Service1 extends _root_.scalapb.grpc.AbstractService {
    override def serviceCompanion = Service1
    /** Computes string length
      */
    def unaryStringLength(request: com.trueaccord.proto.e2e.service.Req1): scala.concurrent.Future[com.trueaccord.proto.e2e.service.Res1]
    // ...
  }
```
```scala
  trait Service1BlockingClient {
    def serviceCompanion = Service1
    /** Computes string length
      */
    def unaryStringLength(request: com.trueaccord.proto.e2e.service.Req1): scala.concurrent.Future[com.trueaccord.proto.e2e.service.Res1]
    // ...
  }
```

### CODE REVIEW

**Comment by thesamet** [comment]
This looks great, @bjaglin  - thanks for sending this over! Will review shortly.

**Comment by bjaglin** [comment]
https://travis-ci.org/scalapb/ScalaPB/jobs/481118610 tells me that `GrpcServicePrinter` is not properly formatted, but when I run `sbt -J-Xmx4500M ++2.11.12 scalafmtCheck test:scalafmtCheck` locally, it looks fine... any idea?

File: compiler-plugin/src/main/scala/scalapb/compiler/DescriptorImplicits.scala [hunk 1]
```diff
@@ -642,6 +673,23 @@ class DescriptorImplicits(params: GeneratorParams, files: Seq[FileDescriptor]) {
 
     def companionExtends: Seq[String] =
       s"_root_.scalapb.GeneratedEnumCompanion[${nameSymbol}]" +: scalaOptions.getCompanionExtendsList.asScala
+
+    def sourcePath: Seq[Int] = {
+      if (enum.isTopLevel) Seq(FileDescriptorProto.ENUM_TYPE_FIELD_NUMBER, enum.getIndex)
+      else
+        enum.getContainingType.sourcePath ++ Seq(
+          DescriptorProto.NESTED_TYPE_FIELD_NUMBER,
```
[the remark this answers is not in the document]
Comment (bjaglin) [reply compiler-plugin/src/main/scala/scalapb/compiler/DescriptorImplicits.scala]: of course, my bad - fixed, re-ran `./make_plugin_proto.sh` and amended the commit

**Comment by bjaglin** [comment]
Just pushed a second commit with the runtime inspection part as you suggested. I had to bootstrap the `scalaDescriptor`s along the way, but I kept it to a minimal since there would be a lot to discuss if we wanted to be exhaustive (particularly how to to model the request/response types and the streaming nature of RPCs).

**Comment by thesamet** [comment]
Awesome, thanks for adding the service/method descriptors. It's a great start, and we can add more APIs later. Can you make the travis tests pass?

**Comment by bjaglin** [comment]
@thesamet I don't see any mention of this in the `CHANGELOG`, is that on purpose because the feature is minor?

**Comment by thesamet** [comment]
Unintentional - can you send a PR?

**Pull request merged**